File: backend/api/schemas/knowledge.py
from __future__ import annotations

from pydantic import BaseModel


class ManualUploadRequest(BaseModel):
    product_id: str
    title: str
    file_type: str


class ManualData(BaseModel):
    id: str
    product_id: str
    title: str
    file_type: str
    status: str
    created_at: str
    updated_at: str


class SpecItem(BaseModel):
    spec_name: str
    spec_value: str
    spec_category: str


class SpecImportRequest(BaseModel):
    product_id: str
    specs: list[SpecItem]


class SpecData(BaseModel):
    id: str
    product_id: str
    spec_name: str
    spec_value: str
    spec_category: str


class SOPCreateRequest(BaseModel):
    product_id: str
    fault_type: str
    fault_category: str
    symptoms: list[str]
    steps: list[str]
    resolution: str


class SOPData(BaseModel):
    id: str
    product_id: str
    fault_type: str
    fault_category: str
    symptoms: list[str]
    steps: list[str]
    resolution: str

File: frontend/src/types/knowledge.ts
export interface ProductManual {
  id: string
  product_id: string
  title: string
  file_type: string
  status: string
  created_at: string
  updated_at: string
}

export interface ProductSpec {
  id: string
  product_id: string
  spec_name: string
  spec_value: string
  spec_category: string
}

export interface FaultSOP {
  id: string
  product_id: string
  fault_type: string
  fault_category: string
  symptoms: Array<{ keyword: string; weight: number }>
  steps: Array<{ order: number; action: string; expected: string; next_if_pass: number; next_if_fail: number }>
  resolution: string
}
